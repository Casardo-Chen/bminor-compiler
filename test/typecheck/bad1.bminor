main: function void() = {
    x:integer = 5;
    if( 4 < 5 )
        if( 10 < 20 )
            for(x = 0; x < 5; x++) 
                if( 4 < "5" )          // Error: Comparing integer with string
                    print "helloo\n";
                else
                    print 5;           // Error: 'print' expects a string, not an integer
        else
            x = "goodbye";             // Error: Assigning a string to an integer variable
}