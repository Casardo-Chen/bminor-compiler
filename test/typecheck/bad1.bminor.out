x resolves to local 0
x resolves to local 0
x resolves to local 0
x resolves to local 0
type error: cannot compare string ("5") to integer (4)
type error: cannot assign string ("goodbye") to integer (x)
3 type error(s) found.
